### PR TITLE
Support Dynamic ModelType

### PR DESCRIPTION
**Pull request by maiqingqiang**
The Swift Package is inconvenient to modify, so I want to change ModelType to dynamic configuration.

Example:

```swift
ModelType.registerModelType("Llama5") { url in
    let configuration = try JSONDecoder().decode(Llama5Configuration.self, from: Data(contentsOf: url))
                return Llama5Model(configuration)
}
```

### CODE REVIEW

**Comment by davidkoski** [comment]
Looks like a nice improvement!

**Review by davidkoski — APPROVED** [review]

File: Libraries/LLM/Tokenizer.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ private func updateTokenizerConfig(_ tokenizerConfig: Config) -> Config {
 }
 
 /// overrides for TokenizerModel/knownTokenizers
-let replacementTokenizers = [
+public var replacementTokenizers = [
```
Comment (davidkoski) [inline Libraries/LLM/Tokenizer.swift:54]: I just realized this isn't thread safe -- we get a warning with all the thread safety checks on and an error in swift 6.  We may need to wrap this up in an `actor`

File: Libraries/LLM/Configuration.swift [hunk 2]
```diff
@@ -26,57 +26,64 @@ public enum StringOrNumber: Codable, Equatable, Sendable {
...
+    public init(rawValue: String) {
+        self.rawValue = rawValue
+    }
+
+    private static func createLlamaModel(url: URL) throws -> LLMModel {
+        let configuration = try JSONDecoder().decode(LlamaConfiguration.self, from: Data(contentsOf: url))
+        return LlamaModel(configuration)
+    }
+
+    private static var creators: [String: (URL) throws -> LLMModel] = [
```
Comment (davidkoski) [inline Libraries/LLM/Configuration.swift:41]: Same thread safety issue here

File: Libraries/LLM/Tokenizer.swift (hunk 1 above)
Comment (maiqingqiang) [reply Libraries/LLM/Tokenizer.swift:54]: Oh my, there is indeed an error warning in Swift 6. If using actors, loadSynchronous may not be synchronous.https://github.com/ml-explore/mlx-swift-examples/blob/cf298551f317e33c390d035b94a3a9433abe9fcc/Libraries/LLM/Load.swift#L50

Comment (davidkoski) [reply Libraries/LLM/Tokenizer.swift:54]: Yes, that is the nature of mutable state -- you need a way to guard it.  We could potentially use a lock here as we don't expect much contention for it (indeed it is likely to be called once per run).  That would let us use unchecked Sendable and keep it synchronous.

Comment (davidkoski) [reply Libraries/LLM/Tokenizer.swift:54]: This has a nice writeup: https://stackoverflow.com/questions/76562442/in-swift-async-await-can-i-use-lock-or-semaphore

We have a nice tight critical section for these maps so something like NSLock should work well.

**Pull request merged**